Tutorials/Level3Tutorial.md

```diff
@@ -0,0 +1,2583 @@
+# Level 3 Tutorial
+
+## Table of Contents
+
+1. [Developing extensions to SST](#DevelopingExtensions)
+2. [Developing Custom SST components](#CustomSSTComponents)
+3. [Developing SST Subcomponents](#DevelopingSubcomponents)
+4. [Connecting Components with Ports](#ConnectingComponents)
+5. [Integration Testing, Integration with PyBind](#IntegrationTesting)
+
+## Developing Extensions to SST <a name="DevelopingExtensions"></a>
+
+### Introduction and Environment
+
+The SST infrastructure provides the ability to develop and utilize 
+component models outside of those provided by the SST-Elements package.  
+SST provides a set of classes for external models to inherit in order 
+to ensure basic interoperability with the SST Core and adjacent models.  
+Externally developed components may implement the core functionality 
+required to model the desired behavior as long as the core SST 
+interfaces are utilized to interact with the SST simulation environment.  
+
+The process by which to develop external component models is as follows:
+1. Outline the directory and file structure for the external component
+2. Implement the simulation component using the interfaces provided by SST
+3. Build the component
+4. Register the component with the SST Core
+5. Integrate the component into a simulation script
+6. Execute the simulation
+
+Using this development workflow, there are a number of key items to keep 
+in mind when implementing external components.
+
+* C++ Standards: The SST Core utilizes C++11 data structures and methods.  External 
+elements should follow the C++11 standard as well.
+* All external elements should implement and register a method that serves as a 
```

Comment:
A clock is not necessary for all components, purely event driven components are fine.

Reply:
fixed in 9feecc2